Bottom: d9233f3e4a7ffcedf846cd2c32244e9fb7bb0059
Top:    691a2c3049cc00eb7ba1e920f782405756529bfb
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-13 02:16:34 +0100

hid/common/hidgl.c: Fix vertex array state preservation

Fixes a bug with crosshair attached objects not being visible
when the grid was rendered.

Reported-by: Colin D Bennett <colin@gibibit.com>
Closes-bug: lp-780958


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index fb6b118..00efdce 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -66,8 +66,6 @@ float global_depth = 0;
 void
 hidgl_init_triangle_array (triangle_buffer *buffer)
 {
-  glEnableClientState (GL_VERTEX_ARRAY);
-  glVertexPointer (3, GL_FLOAT, 0, buffer->triangle_array);
   buffer->triangle_count = 0;
   buffer->coord_comp_count = 0;
 }
@@ -78,7 +76,11 @@ hidgl_flush_triangles (triangle_buffer *buffer)
   if (buffer->triangle_count == 0)
     return;
 
+  glEnableClientState (GL_VERTEX_ARRAY);
+  glVertexPointer (3, GL_FLOAT, 0, buffer->triangle_array);
   glDrawArrays (GL_TRIANGLES, 0, buffer->triangle_count * 3);
+  glDisableClientState (GL_VERTEX_ARRAY);
+
   buffer->triangle_count = 0;
   buffer->coord_comp_count = 0;
 }
